chemprop/featurizers/molgraph/base.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+from collections.abc import Callable
+
+import numpy as np
+from rdkit import Chem
+
+from chemprop.featurizers.molgraph.molgraph import MolGraph
+
+T = TypeVar("T", Chem.Mol, tuple[Chem.Mol, Chem.Mol])
+
+
+class MolGraphFeaturizer(Generic[T], Callable):
+    """A :class:`MolGraphFeaturizer` featurizes input molecules or reactions
+    into :class:`MolGraph`\s"""
+
+    @abstractmethod
+    def __call__(
+        self,
+        input: T,
+        atom_features_extra: np.ndarray | None = None,
+        bond_features_extra: np.ndarray | None = None,
+    ) -> MolGraph:
+        """Featurize the input into a molecular graph
+
+        Parameters
+        ----------
+        input : T
```

Comment:
`T` is an abstract type and is a common notation in programming. There's no need to actually constrain it as I defined it above. IMO it should not be documented.

Reply:
`SimpleMoleculeMolGraphFeaturizer.__call__` will inherit this docstring. It seems like a lot of wasted documentation just to change a single type. The reason `CondensedGraphOfReactionFeaturizer` overwrite this docstring is because we need to note that the additional arguments are unused.